### PR TITLE
Add Spherical ROM as Default in Unit Tests

### PR DESCRIPTION
**Pull request by RealZiangLiu**
* Add spherical ROM model
* Use spherical ROM as default in unit tests
* Add standalone unit test for ROM models (currently only spherical rom)
* Upload cache folder for computed FK results

### CODE REVIEW

File: src/multitask_personalization/rom/models.py [hunk 1]
```diff
@@ -196,3 +197,82 @@ def _visualize_reachable_points(
...
+        seed: int = 0,
+        radius: float = 0.5,
+    ) -> None:
+        super().__init__()
+        self._radius = radius
+        # Create human
+        # Uncomment for debugging
+        # from pybullet_helpers.gui import create_gui_connection
+        # self._physics_client_id = create_gui_connection()
+        self._physics_client_id = p.connect(p.DIRECT)
```
Comment (tomsilver) [inline src/multitask_personalization/rom/models.py:217]: let's put this common stuff into `ROMModel`?

File: src/multitask_personalization/rom/models.py [hunk 2]
```diff
@@ -196,3 +197,82 @@ def _visualize_reachable_points(
...
+        return distance < self._radius + padding
+
+    def sample_reachable_position(self, rng: np.random.Generator) -> NDArray:
+        return np.array(sample_spherical(self._sphere_center, self._radius, rng))
+
+    def get_reachable_joints(self) -> NDArray:
+        raise NotImplementedError("SphericalROMModel does not have reachable joints.")
+
+    def get_reachable_points(self) -> list[NDArray]:
+        raise NotImplementedError("SphericalROMModel does not have reachable joints.")
```
Comment (tomsilver) [inline src/multitask_personalization/rom/models.py:242]: reachable points*

File: src/multitask_personalization/rom/models.py [hunk 3]
```diff
@@ -196,3 +197,82 @@ def _visualize_reachable_points(
...
+    def check_position_reachable(
+        self, position: NDArray, padding: float = 1e-6
+    ) -> bool:
+        distance = float(np.linalg.norm(position - self._sphere_center))
+        return distance < self._radius + padding
+
+    def sample_reachable_position(self, rng: np.random.Generator) -> NDArray:
+        return np.array(sample_spherical(self._sphere_center, self._radius, rng))
+
+    def get_reachable_joints(self) -> NDArray:
```
Comment (tomsilver) [inline src/multitask_personalization/rom/models.py:238]: if not all ROMModels will implement these methods, should we remove them from the ROMModel definition itself (and make them private methods for ROMModels that do implement them)?

File: src/multitask_personalization/rom/models.py [hunk 4]
```diff
@@ -196,3 +197,82 @@ def _visualize_reachable_points(
...
+    def get_reachable_points(self) -> list[NDArray]:
+        raise NotImplementedError("SphericalROMModel does not have reachable joints.")
+
+    def _sample_spherical_points(self, n: int = 500) -> list[NDArray]:
+        return [
+            np.array(sample_spherical(self._sphere_center, self._radius, self._rng))
+            for _ in range(n)
+        ]
+
+    def _visualize_reachable_points(
```
Comment (tomsilver) [inline src/multitask_personalization/rom/models.py:250]: we should also be able to move this code into the parent class I think

**Review by tomsilver — COMMENTED** [review]
thanks a lot! see comments

**Comment by RealZiangLiu** [comment]
Requested fixes all done. One thing though, is there anything in `test_pybullet_csp.py` that's not deterministic? I'm observing inconsistent results between different runs.

**Review by tomsilver — APPROVED** [review]
LGTM thanks! Messaged you about the unit test inconsistency

**Pull request merged**